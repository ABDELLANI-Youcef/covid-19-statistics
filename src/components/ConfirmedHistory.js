import PropTypes from 'prop-types';
import React from 'react';
import styles from '../styles/History.module.css';

const ConfirmedHistory = ({ country }) => {
  const { dates } = country;
  const dataList = Object.keys(dates).splice(0, 30).map(d => (
    <li key={d}>
      {d}
      :
      {' '}
      <span className={styles.number}>
        {dates[d]}
      </span>
      {' '}
    </li>
  ));
  return (
    <div className={styles.history}>
      <h2>Confirmed cases in the last 30 days</h2>
      <ul>
        {dataList}
      </ul>
    </div>
  );
};
ConfirmedHistory.propTypes = {
  country: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default ConfirmedHistory;
