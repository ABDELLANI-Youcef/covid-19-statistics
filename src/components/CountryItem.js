import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from '../styles/CountryItem.module.css';

function CountryItem({ country, name }) {
  return (
    <div className={styles.listing}>
      <p>
        <Link to={{ pathname: `/country/${name}` }}>{ name }</Link>
      </p>
      <p>
        Confirmed cases:
        {' '}
        <span className={styles.number}>
          {country.confirmed}
        </span>
      </p>
      <p>
        Deaths:
        {' '}
        <span className={styles.number}>
          {country.deaths}
        </span>
      </p>
      <p>
        Recovery:
        {' '}
        <span className={styles.number}>
          {country.recovered}
        </span>
      </p>

    </div>
  );
}

CountryItem.propTypes = {
  country: PropTypes.objectOf(PropTypes.any).isRequired,
  name: PropTypes.string.isRequired,
};

export default CountryItem;
