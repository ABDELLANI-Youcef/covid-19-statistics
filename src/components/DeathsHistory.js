import PropTypes from 'prop-types';
import React from 'react';
import styles from '../styles/History.module.css';

const DeathsHistory = ({ country }) => {
  const { dates } = country;
  const dataList = Object.keys(dates).splice(0, 30).map(d => (
    <li key={d}>
      {d}
      :
      {' '}
      <span className={styles.number}>
        {dates[d]}
      </span>
      {' '}
    </li>
  ));
  return (
    <div className={styles.history}>
      <h2>Deaths in the last 30 days</h2>
      <ul>
        {dataList}
      </ul>
    </div>
  );
};
DeathsHistory.propTypes = {
  country: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default DeathsHistory;
