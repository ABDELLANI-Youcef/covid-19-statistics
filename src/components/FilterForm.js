import PropTypes from 'prop-types';

const continents = ['All', 'Africa'];
const FilterForm = ({ clickHandle, filter }) => {
  const changeHandle = e => {
    clickHandle(e.target.value);
  };
  return (
    <form>
      <select value={filter} onChange={changeHandle}>
        {continents.map(con => (
          <option value={con} key={con}>{con}</option>
        ))}
      </select>
    </form>
  );
};

FilterForm.propTypes = {
  clickHandle: PropTypes.func.isRequired,
  filter: PropTypes.PropTypes.string.isRequired,
};

export default FilterForm;
