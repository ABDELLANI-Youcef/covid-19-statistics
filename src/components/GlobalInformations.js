import PropTypes from 'prop-types';
import React from 'react';
import styles from '../styles/GlobalInformations.module.css';

const GlobalInformations = ({ world }) => (
  <div className={styles.container}>
    <div>
      Population in the world:
      {' '}
      <span className={styles.number}>
        {world.population}
      </span>
    </div>
    <div>
      Confirmed cases in the world:
      {' '}
      <span className={styles.number}>
        {world.confirmed}
      </span>
    </div>
    <div>
      Deaths in the world:
      {' '}
      <span className={styles.number}>
        {world.deaths}
      </span>
    </div>
    <div>
      Total of recoveries in the world:
      {' '}
      <span className={styles.number}>
        {world.recovered}
      </span>
    </div>
  </div>

);

GlobalInformations.propTypes = {
  world: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default GlobalInformations;
