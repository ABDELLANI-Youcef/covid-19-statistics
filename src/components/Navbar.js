import { Link } from 'react-router-dom';
import styles from './Navbar.module.css';

const Navbar = () => (
  <div className={styles.navbar}>
    <Link to="/">Home</Link>
    <h1>COVID-19 AROUND THE WORLD</h1>
    <span>Get the last numbers of cases, deaths, and  recovering</span>
  </div>
);

export default Navbar;
