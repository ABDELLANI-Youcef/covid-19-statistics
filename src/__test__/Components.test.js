import React from 'react';
import renderer from 'react-test-renderer';
import {
  BrowserRouter,
} from 'react-router-dom';
import Navbar from '../components/Navbar';
import FilterForm from '../components/FilterForm';
import GlobalDetail from '../components/GeneralDetail';
import GlobalInformations from '../components/GlobalInformations';
import CountryItem from '../components/CountryItem';
import DeathsHistory from '../components/DeathsHistory';
import ConfirmedHistory from '../components/ConfirmedHistory';

it('should render the Filter Form correctly', () => {
  const filter = {
    continent: 'All',
    maxCases: -1,
    minCases: 0,
    maxDeaths: -1,
    minDeaths: 0,
  };
  const navbar = renderer.create(
    <FilterForm filter={filter} clickHandle={() => true} />,
  ).toJSON();
  expect(navbar).toMatchSnapshot();
});

it('should render the Navbar correctly', () => {
  const navbar = renderer.create(
    <BrowserRouter>
      <Navbar />
    </BrowserRouter>,
  ).toJSON();
  expect(navbar).toMatchSnapshot();
});

it('should return the GlobalDetail correctly', () => {
  const algeriaInfons = {
    confirmed: 115265,
    recovered: 79887,
    deaths: 3036,
    country: 'Algeria',
    population: 41318142,
    sq_km_area: 2381741,
    life_expectancy: '69.7',
    elevation_in_meters: 800,
    continent: 'Africa',
    abbreviation: 'DZ',
    location: 'Northern Africa',
    iso: 12,
    capital_city: 'Alger',
    lat: '28.0339',
    long: '1.6596',
    updated: '2021/03/15 18:27:18+00',
  };

  const globalDetail = renderer.create(
    <GlobalDetail country={algeriaInfons} />,
  ).toJSON();
  expect(globalDetail).toMatchSnapshot();
});

it('should return GlobalInformations correctly', () => {
  const world = {
    population: 7444509223,
    confirmed: 120096586,
    recovered: 68065974,
    deaths: 2657501,
  };
  const globalInformations = renderer.create(
    <GlobalInformations world={world} />,
  ).toJSON();
  expect(globalInformations).toMatchSnapshot();
});

it('should return the CountryItem correctly', () => {
  const algeriaInfons = {
    confirmed: 115265,
    recovered: 79887,
    deaths: 3036,
    country: 'Algeria',
    population: 41318142,
    sq_km_area: 2381741,
    life_expectancy: '69.7',
    elevation_in_meters: 800,
    continent: 'Africa',
    abbreviation: 'DZ',
    location: 'Northern Africa',
    iso: 12,
    capital_city: 'Alger',
    lat: '28.0339',
    long: '1.6596',
    updated: '2021/03/15 18:27:18+00',
  };
  const countryItem = renderer.create(
    <BrowserRouter>
      <CountryItem country={algeriaInfons} name="Algeria" />
    </BrowserRouter>,
  ).toJSON();
  expect(countryItem).toMatchSnapshot();
});

it('should return DeathsHistory correctly', () => {
  const country = {
    dates: {
      '2021-03-14': 3036,
      '2021-03-13': 3034,
      '2021-03-12': 3031,
      '2021-03-11': 3028,
      '2021-03-10': 3026,
      '2021-03-09': 3022,
      '2021-03-08': 3018,
      '2021-03-07': 3013,
      '2021-03-06': 3010,
      '2021-03-05': 3007,
      '2021-03-04': 3002,
      '2021-03-03': 2996,
      '2021-03-02': 2991,
      '2021-03-01': 2987,
      '2021-02-28': 2983,
      '2021-02-27': 2979,
      '2021-02-26': 2977,
      '2021-02-25': 2973,
      '2021-02-24': 2970,
      '2021-02-23': 2967,
      '2021-02-22': 2964,
      '2021-02-21': 2961,
      '2021-02-20': 2958,
      '2021-02-19': 2954,
      '2021-02-18': 2950,
      '2021-02-17': 2947,
      '2021-02-16': 2945,
      '2021-02-15': 2943,
      '2021-02-14': 2939,
      '2021-02-13': 2935,
    },
  };

  const deathsHistory = renderer.create(
    <DeathsHistory country={country} />,
  ).toJSON();
  expect(deathsHistory).toMatchSnapshot();
});

it('should return ConfirmedHistory correctly', () => {
  const country = {
    dates: {
      '2021-03-14': 115265,
      '2021-03-13': 115143,
      '2021-03-12': 115008,
      '2021-03-11': 114851,
      '2021-03-10': 114681,
      '2021-03-09': 114543,
      '2021-03-08': 114382,
      '2021-03-07': 114234,
      '2021-03-06': 114104,
      '2021-03-05': 113948,
      '2021-03-04': 113761,
      '2021-03-03': 113593,
      '2021-03-02': 113430,
      '2021-03-01': 113255,
      '2021-02-28': 113092,
      '2021-02-27': 112960,
      '2021-02-26': 112805,
      '2021-02-25': 112622,
      '2021-02-24': 112461,
      '2021-02-23': 112279,
      '2021-02-22': 112094,
      '2021-02-21': 111917,
      '2021-02-20': 111764,
      '2021-02-19': 111600,
      '2021-02-18': 111418,
      '2021-02-17': 111247,
      '2021-02-16': 111069,
      '2021-02-15': 110894,
      '2021-02-14': 110711,
      '2021-02-13': 110513,
    },
  };

  const confirmedHistory = renderer.create(
    <ConfirmedHistory country={country} />,
  ).toJSON();
  expect(confirmedHistory).toMatchSnapshot();
});
