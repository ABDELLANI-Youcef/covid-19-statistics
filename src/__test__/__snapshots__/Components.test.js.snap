// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Filter Form correctly 1`] = `
<form
  className="filterForm"
>
  <label
    htmlFor="continent"
  >
    Continent:
    <select
      id="continent"
      name="continent"
      onChange={[Function]}
      value="All"
    >
      <option
        value="All"
      >
        All
      </option>
      <option
        value="Africa"
      >
        Africa
      </option>
      <option
        value="Asia"
      >
        Asia
      </option>
      <option
        value="North America"
      >
        North America
      </option>
      <option
        value="South America"
      >
        South America
      </option>
      <option
        value="Oceania"
      >
        Oceania
      </option>
      <option
        value="Europe"
      >
        Europe
      </option>
    </select>
  </label>
  <div
    className="infoType"
  >
    <label
      htmlFor="min-Cases"
    >
      <span
        className="input_label"
      >
        Minimun number of cases:
      </span>
      <input
        id="min-Cases"
        name="min-Cases"
        onChange={[Function]}
        type="number"
        value={0}
      />
    </label>
    <label
      htmlFor="max-Cases"
    >
      <span
        className="input_label"
      >
        Maximun number of cases:
      </span>
      <input
        id="max-Cases"
        name="max-Cases"
        onChange={[Function]}
        type="number"
        value={-1}
      />
    </label>
  </div>
  <div
    className="infoType"
  >
    <label
      htmlFor="min-Deaths"
    >
      <span
        className="input_label"
      >
        Minimun number of deaths:
      </span>
      <input
        id="min-Deaths"
        name="min-Deaths"
        onChange={[Function]}
        type="number"
        value={0}
      />
    </label>
    <label
      htmlFor="max-Deaths"
    >
      <span
        className="input_label"
      >
        Maximun number of deaths:
      </span>
      <input
        id="max-Deaths"
        name="max-Deaths"
        onChange={[Function]}
        type="number"
        value={-1}
      />
    </label>
  </div>
</form>
`;

exports[`should render the Navbar correctly 1`] = `
<div
  className="navbar"
>
  <a
    href="/"
    onClick={[Function]}
  >
    Home
  </a>
  <h1>
    COVID-19 AROUND THE WORLD
  </h1>
  <span>
    Get the last numbers of cases, deaths, and  recovering
  </span>
</div>
`;

exports[`should return ConfirmedHistory correctly 1`] = `
<div
  className="history"
>
  <h2>
    Confirmed cases in the last 30 days
  </h2>
  <ul>
    <li>
      2021-03-14
      :
       
      <span
        className="number"
      >
        115265
      </span>
       
    </li>
    <li>
      2021-03-13
      :
       
      <span
        className="number"
      >
        115143
      </span>
       
    </li>
    <li>
      2021-03-12
      :
       
      <span
        className="number"
      >
        115008
      </span>
       
    </li>
    <li>
      2021-03-11
      :
       
      <span
        className="number"
      >
        114851
      </span>
       
    </li>
    <li>
      2021-03-10
      :
       
      <span
        className="number"
      >
        114681
      </span>
       
    </li>
    <li>
      2021-03-09
      :
       
      <span
        className="number"
      >
        114543
      </span>
       
    </li>
    <li>
      2021-03-08
      :
       
      <span
        className="number"
      >
        114382
      </span>
       
    </li>
    <li>
      2021-03-07
      :
       
      <span
        className="number"
      >
        114234
      </span>
       
    </li>
    <li>
      2021-03-06
      :
       
      <span
        className="number"
      >
        114104
      </span>
       
    </li>
    <li>
      2021-03-05
      :
       
      <span
        className="number"
      >
        113948
      </span>
       
    </li>
    <li>
      2021-03-04
      :
       
      <span
        className="number"
      >
        113761
      </span>
       
    </li>
    <li>
      2021-03-03
      :
       
      <span
        className="number"
      >
        113593
      </span>
       
    </li>
    <li>
      2021-03-02
      :
       
      <span
        className="number"
      >
        113430
      </span>
       
    </li>
    <li>
      2021-03-01
      :
       
      <span
        className="number"
      >
        113255
      </span>
       
    </li>
    <li>
      2021-02-28
      :
       
      <span
        className="number"
      >
        113092
      </span>
       
    </li>
    <li>
      2021-02-27
      :
       
      <span
        className="number"
      >
        112960
      </span>
       
    </li>
    <li>
      2021-02-26
      :
       
      <span
        className="number"
      >
        112805
      </span>
       
    </li>
    <li>
      2021-02-25
      :
       
      <span
        className="number"
      >
        112622
      </span>
       
    </li>
    <li>
      2021-02-24
      :
       
      <span
        className="number"
      >
        112461
      </span>
       
    </li>
    <li>
      2021-02-23
      :
       
      <span
        className="number"
      >
        112279
      </span>
       
    </li>
    <li>
      2021-02-22
      :
       
      <span
        className="number"
      >
        112094
      </span>
       
    </li>
    <li>
      2021-02-21
      :
       
      <span
        className="number"
      >
        111917
      </span>
       
    </li>
    <li>
      2021-02-20
      :
       
      <span
        className="number"
      >
        111764
      </span>
       
    </li>
    <li>
      2021-02-19
      :
       
      <span
        className="number"
      >
        111600
      </span>
       
    </li>
    <li>
      2021-02-18
      :
       
      <span
        className="number"
      >
        111418
      </span>
       
    </li>
    <li>
      2021-02-17
      :
       
      <span
        className="number"
      >
        111247
      </span>
       
    </li>
    <li>
      2021-02-16
      :
       
      <span
        className="number"
      >
        111069
      </span>
       
    </li>
    <li>
      2021-02-15
      :
       
      <span
        className="number"
      >
        110894
      </span>
       
    </li>
    <li>
      2021-02-14
      :
       
      <span
        className="number"
      >
        110711
      </span>
       
    </li>
    <li>
      2021-02-13
      :
       
      <span
        className="number"
      >
        110513
      </span>
       
    </li>
  </ul>
</div>
`;

exports[`should return DeathsHistory correctly 1`] = `
<div
  className="history"
>
  <h2>
    Deaths in the last 30 days
  </h2>
  <ul>
    <li>
      2021-03-14
      :
       
      <span
        className="number"
      >
        3036
      </span>
       
    </li>
    <li>
      2021-03-13
      :
       
      <span
        className="number"
      >
        3034
      </span>
       
    </li>
    <li>
      2021-03-12
      :
       
      <span
        className="number"
      >
        3031
      </span>
       
    </li>
    <li>
      2021-03-11
      :
       
      <span
        className="number"
      >
        3028
      </span>
       
    </li>
    <li>
      2021-03-10
      :
       
      <span
        className="number"
      >
        3026
      </span>
       
    </li>
    <li>
      2021-03-09
      :
       
      <span
        className="number"
      >
        3022
      </span>
       
    </li>
    <li>
      2021-03-08
      :
       
      <span
        className="number"
      >
        3018
      </span>
       
    </li>
    <li>
      2021-03-07
      :
       
      <span
        className="number"
      >
        3013
      </span>
       
    </li>
    <li>
      2021-03-06
      :
       
      <span
        className="number"
      >
        3010
      </span>
       
    </li>
    <li>
      2021-03-05
      :
       
      <span
        className="number"
      >
        3007
      </span>
       
    </li>
    <li>
      2021-03-04
      :
       
      <span
        className="number"
      >
        3002
      </span>
       
    </li>
    <li>
      2021-03-03
      :
       
      <span
        className="number"
      >
        2996
      </span>
       
    </li>
    <li>
      2021-03-02
      :
       
      <span
        className="number"
      >
        2991
      </span>
       
    </li>
    <li>
      2021-03-01
      :
       
      <span
        className="number"
      >
        2987
      </span>
       
    </li>
    <li>
      2021-02-28
      :
       
      <span
        className="number"
      >
        2983
      </span>
       
    </li>
    <li>
      2021-02-27
      :
       
      <span
        className="number"
      >
        2979
      </span>
       
    </li>
    <li>
      2021-02-26
      :
       
      <span
        className="number"
      >
        2977
      </span>
       
    </li>
    <li>
      2021-02-25
      :
       
      <span
        className="number"
      >
        2973
      </span>
       
    </li>
    <li>
      2021-02-24
      :
       
      <span
        className="number"
      >
        2970
      </span>
       
    </li>
    <li>
      2021-02-23
      :
       
      <span
        className="number"
      >
        2967
      </span>
       
    </li>
    <li>
      2021-02-22
      :
       
      <span
        className="number"
      >
        2964
      </span>
       
    </li>
    <li>
      2021-02-21
      :
       
      <span
        className="number"
      >
        2961
      </span>
       
    </li>
    <li>
      2021-02-20
      :
       
      <span
        className="number"
      >
        2958
      </span>
       
    </li>
    <li>
      2021-02-19
      :
       
      <span
        className="number"
      >
        2954
      </span>
       
    </li>
    <li>
      2021-02-18
      :
       
      <span
        className="number"
      >
        2950
      </span>
       
    </li>
    <li>
      2021-02-17
      :
       
      <span
        className="number"
      >
        2947
      </span>
       
    </li>
    <li>
      2021-02-16
      :
       
      <span
        className="number"
      >
        2945
      </span>
       
    </li>
    <li>
      2021-02-15
      :
       
      <span
        className="number"
      >
        2943
      </span>
       
    </li>
    <li>
      2021-02-14
      :
       
      <span
        className="number"
      >
        2939
      </span>
       
    </li>
    <li>
      2021-02-13
      :
       
      <span
        className="number"
      >
        2935
      </span>
       
    </li>
  </ul>
</div>
`;

exports[`should return GlobalInformations correctly 1`] = `
<div
  className="container"
>
  <div>
    Population in the world:
     
    <span
      className="number"
    >
      7444509223
    </span>
  </div>
  <div>
    Confirmed cases in the world:
     
    <span
      className="number"
    >
      120096586
    </span>
  </div>
  <div>
    Deaths in the world:
     
    <span
      className="number"
    >
      2657501
    </span>
  </div>
  <div>
    Total of recovered Persons in the world:
     
    <span
      className="number"
    >
      68065974
    </span>
  </div>
</div>
`;

exports[`should return the CountryItem correctly 1`] = `
<div
  className="listing"
>
  <p>
    <a
      href="/country/Algeria"
      onClick={[Function]}
    >
      Algeria
    </a>
  </p>
  <p>
    Confirmed cases:
     
    <span
      className="number"
    >
      115265
    </span>
  </p>
  <p>
    Deaths:
     
    <span
      className="number"
    >
      3036
    </span>
  </p>
  <p>
    Recovery:
     
    <span
      className="number"
    >
      79887
    </span>
  </p>
</div>
`;

exports[`should return the GlobalDetail correctly 1`] = `
<div
  className="container"
>
  <div
    className="information"
  >
    The capital:
     
    <span
      className="number"
    >
      Alger
    </span>
  </div>
  <div
    className="information"
  >
    Population:
     
    <span
      className="number"
    >
      41318142
    </span>
  </div>
  <div
    className="information"
  >
    confirmed:
     
    <span
      className="number danger"
    >
      115265
    </span>
     
    confirmed case
  </div>
  <div
    className="information"
  >
    <span
      className="number danger"
    >
      3036
    </span>
     
    deaths
  </div>
  <div
    className="information"
  >
    <span
      className="number safe"
    >
      79887
    </span>
     
    recovered persons
  </div>
</div>
`;
